File: lib/features/contacts/data/contact_service.dart
import 'package:dio/dio.dart';
import 'package:try_neuro/core/network/http_client.dart';
import 'package:try_neuro/features/contacts/domain/contact_model.dart';
import 'package:try_neuro/service_locator.dart';

class ContactService {
  final Dio _dio = sl<HttpClient>().dio;

  /// Получение списка контактов с пагинацией.
  /// Возвращает Map с 'contacts' (List) и 'isLast' (bool).
  Future<Map<String, dynamic>> getContactsPaged({String? query, int page = 0, int size = 20}) async {
    final response = await _dio.get('/contacts', queryParameters: {
      if (query != null && query.isNotEmpty) 'query': query,
      'page': page,
      'size': size,
    });

    final List<dynamic> content = response.data['content'];
    final bool isLast = response.data['last'] ?? true;

    return {
      'contacts': content.map((json) => Contact.fromJson(json)).toList(),
      'isLast': isLast,
    };
  }

  // Оставляем для совместимости (используется в других местах)
  Future<List<Contact>> getContacts({String? query}) async {
    final result = await getContactsPaged(query: query, size: 50);
    return result['contacts'] as List<Contact>;
  }

  Future<Contact?> getContactById(String id) async {
    try {
      final response = await _dio.get('/contacts/$id');
      return Contact.fromJson(response.data);
    } catch (e) {
      return null;
    }
  }

  Future<int> getContactsCount() async {
    final response = await _dio.get('/contacts/count');
    return response.data as int;
  }

  Future<Contact?> findContactByPhone(String phone) async {
    try {
      final response = await _dio.get('/contacts/by-phone', queryParameters: {'phone': phone});
      return Contact.fromJson(response.data);
    } catch (e) {
      return null;
    }
  }

  Future<Contact> addContact({required String name, required List<String> phones, String? email, String? notes}) async {
    final response = await _dio.post('/contacts', data: {
      'name': name,
      'phones': phones,
      'email': email,
      'notes': notes,
    });
    return Contact.fromJson(response.data);
  }

  Future<Contact> updateContact(Contact contact) async {
    final response = await _dio.put('/contacts/${contact.id}', data: contact.toJson());
    return Contact.fromJson(response.data);
  }

  Future<void> deleteContact(String id) async {
    await _dio.delete('/contacts/$id');
  }
}

File: frontend-svelte/src/lib/services/contactService.js
import api from '$lib/api.js';

export const contactService = {
    // Возвращаем Page объект напрямую (content, totalElements и т.д.)
    async getContacts(query = '', showAll = false, page = 0, size = 25) {
        const response = await api.get('/admin/clients', {
            params: {
                query: query.trim(),
                showAll: showAll,
                page,
                size
            }
        });
        return response.data;
    },

    async findContactByPhone(phone) {
        const clean = phone.replace(/\D/g, '');
        try {
            // Используем админский путь для поиска
            const response = await api.get('/admin/clients/by-phone', { params: { phone: clean } });
            return response.data;
        } catch (e) {
            return null;
        }
    },

    async getContactById(id) {
        const response = await api.get(`/admin/clients/${id}`);
        return response.data;
    },

    async addContact(contact) {
        const response = await api.post('/admin/clients', contact);
        return response.data;
    },

    async updateContact(id, contact) {
        const response = await api.put(`/admin/clients/${id}`, contact);
        return response.data;
    },

    async deleteContact(id) {
        await api.delete(`/admin/clients/${id}`);
    },

    async exportContacts(query = '', showAll = false) {
        const response = await api.get('/admin/clients/export', {
            params: {
                query: query.trim(),
                showAll: showAll
            },
            responseType: 'blob'
        });
        return response.data;
    }
};
